drafts/Asynchronous-programming-made-resilient-with-MPFT.adoc

```diff
@@ -0,0 +1,230 @@
+
+But there's one in particular which is going to help you make your asynchronous methods resilient to faults: https://download.eclipse.org/microprofile/microprofile-fault-tolerance-2.1/apidocs/org/eclipse/microprofile/faulttolerance/Asynchronous.html[@Asynchronous].
+
+Let's show you what `@Asynchronous` does and how it does it.
+
+== What does `@Asynchronous` do?
+The `@Asynchronous` annotation makes your method run asynchronously, meaning that rather than running straightaway on the main thread when you call it, it will run some time later, usually on another thread. To do this, we need to add the `@Asynchronous` annotation to the method, and we change the method to return either a https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html[CompletionStage] (preferable) or a https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html[Future]. In our method body, we can use `CompletableFuture.completedFuture()` to make our result compatible with the new return type.
+
+[source,java]
+----
+@Asynchronous
+public CompletionStage<String> serviceA() { 
+
+	return CompletableFuture.completedFuture("serviceA");
+
+}
+----
+
+What does this do when we call the method?
+
+Well, usually when you call a method, you call it, and then it returns a result. However, when we use the `@Asynchronous` annotation, Fault Tolerance gets in between. When you call the method, Fault Tolerance first schedules your method to run later on a different thread, then it returns you a `CompletionStage` or `Future`. This isn't the `CompletionStage` or `Future` that's returned from your method, because your method hasn't run yet, it's an object created entirely by Fault Tolerance. 
+
+image::/img/blog/FT-basic-asynchronous-execution.png[Fault Tolerance asynchronous execution flow]
+
+Sometime later, your method runs and returns a `CompletionStage` or `Future`. Now, Fault Tolerance will go and complete the `CompletionStage` or `Future` object which was returned to the caller earlier, with the value that was returned from your method. You might be wondering *_"What if the returned result is an exception? What can I do?"_*.  Read along.
+
+== How you can use `@Asynchronous`
+So `@Asynchronous` sounds great, right? But how are we going to use it to make our asynchronous programs resilient? Let's go through 2 use-cases:
+
+1. Applying Fault Tolerance to an API which is already asynchronous
+2. Using `@Asynchronous` annotation to do two things in parallel with resilience
+
+=== 1. Applying to An Already Asynchronous API
```

Comment:
Asynchronous API -> Asynchronous Method

Reply:
I'm happy to make this change, but think `method` should not be capitalised: https://www.oracle.com/java/technologies/glossary.html

What do you think?

Reply:
@Joseph-Cass If the heading is title cased (as it currently is), then "method" would be capitalised as a normal word.

However, it looks like the other headings in this article aren't title cased, I think this is just left over from when you were referring to it as AAAAA, so you should probably decapitalise everything for consistency.

Reply:
Ah yes, correct